src/components/ChapterRequest.tsx

```diff
@@ -92,165 +84,44 @@ const ChapterRequest = (props: ChapterRequestProps) => {
           {qas.map((question) => (
             <MoreInformation key={question.question} {...question} />
           ))}
-          <div className="mt-2 flex flex-row space-x-2">
-            <div
-              className="w-1/2 cursor-pointer rounded-xl bg-dark-teal py-1 text-center text-white hover:bg-[#1b4448]"
-              onClick={async () => {
-                handleChapterRequest({
-                  body: {
-                    chapterRequestId: request.id,
-                    approved: true,
-                  },
-                }).then((res) => {
-                  if (res.code === "SUCCESS") {
-                    router.refresh();
-                  } else {
-                    alert("Please refresh the page and try again");
-                  }
-                });
-              }}
-            >
-              Accept
+          {!fetching ? (
+            <div className="mt-2 flex flex-row space-x-2">
+              <div
+                className="w-1/2 cursor-pointer rounded-xl bg-dark-teal py-1 text-center text-white hover:bg-[#1b4448]"
+                onClick={() =>
+                  throttleChapterRequest({
+                    body: {
+                      chapterRequestId: request.id,
+                      approved: true,
+                    },
+                  })
+                }
+              >
+                Accept
+              </div>
+              <div
+                className="w-1/2 cursor-pointer rounded-xl bg-sunset-orange py-1 text-center text-white hover:bg-[#ED5858]"
+                onClick={async () =>
+                  throttleChapterRequest({
+                    body: {
+                      chapterRequestId: request.id,
+                      approved: false,
+                    },
+                  })
+                }
+              >
+                Decline
+              </div>
             </div>
-            <div
-              className="w-1/2 cursor-pointer rounded-xl bg-sunset-orange py-1 text-center text-white hover:bg-[#ED5858]"
-              onClick={async () => {
-                handleChapterRequest({
-                  body: {
-                    chapterRequestId: request.id,
-                    approved: false,
-                  },
-                }).then((res) => {
-                  if (res.code === "SUCCESS") {
-                    router.refresh();
-                  } else {
-                    alert("Please refresh the page and try again");
```

Comment:
Looks like we opt out of the alert when the request encounters an an error, maybe it's better to keep so that users get that feedback?

Reply:
I think I was lazy to handle the cases. But you're probably right that we should handle errors.